scripts/processAndDownloadContracts.ts

```diff
@@ -17,21 +17,25 @@ async function processContracts() {
 			...await getAaveV2Misc()
 		]
 	const output = `
-
-const contractData: [string, string, string, string | null][] = ${JSON.stringify(contractData.map((x) => [addressString(x.address), x.data.name, x.data.protocol, x.data.logoURI]), null, 1)}
+import * as contractData from './contractMetadata.json';
 export type ContractDefinition = {
-    name: string,
-    logoURI: string | undefined,
-    protocol: string | undefined
+	name: string,
+	logoURI: string | undefined,
+	protocol: string | undefined
 }
 export const contractMetadata = new Map<string, ContractDefinition>(
-    contractData.map( ([address, name, protocol, logoURI] : [string, string, string, string | null]) => [address, {
-      name: name,
-      protocol: protocol,
-      logoURI: logoURI === null ? undefined : logoURI
-    }])
+	contractData.reduce(( acc, [address, name, protocol, logoURI] ) => {
+		if (!address || !name) return acc
+		return acc.concat([
+			[address, {
+				name: name,
+				protocol: protocol === null ? undefined : protocol,
+				logoURI: logoURI === null ? undefined : logoURI
+			}]])
+	}, [] as [string, ContractDefinition][])
 )
 `
+	fs.writeFileSync(`${OUTPUT_SRC_DIR}/contractMetadata.json`, JSON.stringify(contractData.map((x) => [addressString(x.address), x.data.name, x.data.protocol, x.data.logoURI]), null, '\t'), 'utf-8')
```

Comment:
It *might* be worth extracting this map function out into a named function to improve readability.  Very borderline case for me.

Reply:
Ya i did it for some of the longer ones, i'll do it here too